### DISCUSSION TITLE
Changing resoluton and framerate

### DISCUSSION THREAD

**Discussion by peteallenm** [roleoroleo/yi-hack-Allwinner-v2#326] [Q&A] [answered]
Hi,

This question is potentially quite off-topic. Apologies if so.
I'm attempting a project which needs to stream video at extremely low latency (aiming for 30ms), and I'm wondering if I can use a security camera as a hardware platform. This makes yi-hack extremely attractive, but I'm struggling to work out what yi-hack has control of and what it doesn't.
What is it that does the streaming on the yi cameras? I notice h264_grabber takes encoded h264 from /dev/shm/fshare_frame_buffer . What puts it there and does the encoding?

Is it possible to modify the resolution and framerate? Ideally I'm looking for 720p/120fps (but can start with 60fps for prototypes).
Another (probably long term better) option is to kill process controls the camera and encoding, and either use gstreamer or write a separate program that accesses the camera, encodes with cedarx (possibly using mjpeg one block at a time) and sends it out over udp.

Thank you for your help.

**Comment by roleoroleo** [comment, marked as answer]
Here:
https://github.com/roleoroleo/yi-hack-Allwinner-v2/discussions/306#discussioncomment-1897950
there is a topic about changing parameters of the cam.

The capture and encode features are managed by the original yi firmware.
Configuration of sensor, configuration of hw encoder, cedarx are not available in a simple manner.
The cpu is not enough to sw encode the stream.
The framebuffer is filled by rmm daemon: a big elf that takes care of everything.

So the control is very poor.

> Another (probably long term better) option is to kill process controls the camera and encoding, and either use gstreamer or write a separate program that accesses the camera, encodes with cedarx (possibly using mjpeg one block at a time) and sends it out over udp.

This could be an option, but I didn't find a "friendly" SDK to try. And it's not my goal.

**Reply by peteallenm** [reply]
Great, thank you the info, that's really helpful

**Comment by roleoroleo** [comment]
About sdk, try to google linux-sunxi.
Or try these:
https://linux-sunxi.org/Main_Page
https://linux-sunxi.org/SDK_build_howto
https://github.com/linux-sunxi
https://github.com/allwinner-dev-team
https://github.com/lindenis-org
